hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/ha/SCMRatisServerImpl.java

```diff
@@ -260,12 +259,8 @@ public List<String> getRatisRoles() throws IOException {
         LOG.error("SCM Ratis PeerInetAddress {} is unresolvable",
             peer.getAddress());
       }
-      boolean isLocal = false;
-      if (peerInetAddress != null) {
-        isLocal = NetUtils.isLocalAddress(peerInetAddress);
-      }
       ratisRoles.add((peer.getAddress() == null ? "" :
-              peer.getAddress().concat(isLocal ?
+          peer.getAddress().concat(peer == getLeader() ?
```

Comment:
Can you please move the `getLeader()` call outside the loop?  I don't think it's necessary to get it for each peer.

Reply:
Also, I think we should use `equals()` instead of `==` for comparing peers.

Reply:
Updated, please have a check.